eth-registration/src/main/kotlin/com/d3/eth/registration/wallet/EthereumRegistrationHelper.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright D3 Ledger, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.d3.eth.registration.wallet
+
+import org.web3j.crypto.ECKeyPair
+import org.web3j.crypto.Hash
+import org.web3j.crypto.Keys
+import org.web3j.crypto.Sign
+
+/**
+ * Create signature proof by signing hash of address generated from public key.
+ * @param ecKeyPair EC keypair
+ * @return proof singed by EC keypair
+ */
+fun createRegistrationProof(ecKeyPair: ECKeyPair): EthereumRegistrationProof {
+    val address = Keys.getAddress(ecKeyPair.publicKey)
+    val esig = ecKeyPair.sign(Hash.sha3(address.toByteArray()))
+    return EthereumRegistrationProof(esig, ecKeyPair.publicKey)
+}
+
+/**
+ * Check proof is actually signature of hash of address get from public key.
+ * @param proof signature and public key
+ * @return true if signature is correct, false otherwise
+ */
+fun checkRegistrationProof(proof: EthereumRegistrationProof): Boolean {
+    val address = Keys.getAddress(proof.publicKey)
+
+    // Iterate recId [0..3] while the correct way not found
+    for (i in 0..3) {
```

Comment:
Why 3?

Reply:
The recId is an index from 0 to 3 which indicates which of the 4 possible keys is the
correct one. Because the key recovery operation yields multiple potential keys, the correct
key must either be stored alongside the
signature, or you must be willing to try each recId in turn until you find one that outputs
the key you are expecting